### PR TITLE
Add docs on storage compaction concurrency

### PR DESCRIPTION
**Pull request by rmloveland**
Fixes:

- DOC-9140
- DOC-9169

Summary of changes:

- Add an include with docs for the `COCKROACH_COMPACTION_CONCURRENCY` environment variable added in v23.2

- Share that include around into a bunch of other pages where compaction concurrency is mentioned

- As part of the general theme of "touching compaction concurrency things", make a bunch of mentions of the term 'compaction' point to the information in the 'Architecture > Storage Layer > Compaction' docs.

- Update a few pre-existing mentions of the `storage.marked-for-compaction-files` metric to be links to the 'metrics' page

### CODE REVIEW

**Comment by rmloveland** [comment]
Hi @jbowens I took a shot at documenting this env var.  In a couple places in the PR I added the text 'XXX' with a broken link and a question for you which is why the docs build is broken

Happy to update as needed, please let me know if anything here is incorrect or missing important information etc - thanks!

File: src/current/_includes/v23.2/storage/compaction-concurrency.md [hunk 1]
```diff
@@ -0,0 +1 @@
+By default, the [storage engine]({% link {{ page.version.version }}/architecture/storage-layer.md %}) uses a compaction concurrency of 4, with one slot dedicated to flushes. If you have sufficient IOPs and CPU headroom, you can consider increasing this setting via the `COCKROACH_COMPACTION_CONCURRENCY` environment variable for increased performance under some workloads. We **strongly recommend** testing your workload against non-default values of this setting.
```
Comment (jbowens) [inline src/current/_includes/v23.2/storage/compaction-concurrency.md:1]: The new `COCKROACH_COMPACTION_CONCURRENCY` environment variable doesn't include flushes (whereas the old undocumented rocksdb one did. So I think we should say "uses a compaction concurrency of 3", without referring to flushes.

File: src/current/v23.2/cluster-setup-troubleshooting.md [hunk 2]
```diff
@@ -174,11 +174,17 @@ The `kv.snapshot_rebalance.max_rate` and `kv.snapshot_recovery.max_rate` [cluste
...
+After [compaction]({% link {{ page.version.version }}/architecture/storage-layer.md %}#compaction) has completed, lower `kv.snapshot_rebalance.max_rate` and `kv.snapshot_recovery.max_rate` to their [default values]({% link {{ page.version.version }}/cluster-settings.md %}). As you add nodes to the cluster, slowly increase both cluster settings, if desired. This will control the rate of new ingestions for newly added nodes. Meanwhile, monitor the cluster for unhealthy increases in [IOPS]({% link {{ page.version.version }}/common-issues-to-monitor.md %}#disk-iops) and [CPU]({% link {{ page.version.version }}/common-issues-to-monitor.md %}#cpu).
+
+{{site.data.alerts.callout_success}}
+[XXX](): <font color="red">QUESTION</font>: the instructions above say to reduce your workload concurrency, so it seems like maybe we should NOT be recommending increasing compaction concurrency here? b/c if we had IOPs and CPU headroom we would not be in this bad state, and thus increasing compaction concurrency would use *even more* IO / CPU right?  That's my sense anyway but please confirm.
```
Comment (jbowens) [inline src/current/v23.2/cluster-setup-troubleshooting.md:184]: If we're falling behind, there's two possibilities:

1. Hardware resources (IO or CPU) are exhausted.
2. Compaction concurrency is exhausted.

It can be hard to differentiate the two, but it's possible for rebalancing to grow the LSM too fast while still not exhausting IO or CPU, because in order to fully utilize the resources we need more concurrent compactions.

To complicate things a little more, once the LSM becomes inverted, you may see high CPU usage as a consequence of the inverted LSM. But it might be the case that if you had increased the compaction concurrency earlier, the LSM may have never become inverted, and CPU usage may have remained within acceptable levels.

I'm not sure what all that means with respect to what we should document.

File: src/current/_includes/v23.2/storage/compaction-concurrency.md (hunk 1 above)
Comment (rmloveland) [reply src/current/_includes/v23.2/storage/compaction-concurrency.md:1]: OK updated to 

> By default, the [storage engine]({% link {{ page.version.version }}/architecture/storage-layer.md %}) uses a compaction concurrency of 3.

File: src/current/v23.2/cluster-setup-troubleshooting.md (hunk 2 above)
Comment (rmloveland) [reply src/current/v23.2/cluster-setup-troubleshooting.md:184]: Ya what you describe does seem quite complicated to understand, especially if one is in the middle of following these other instructions that seem to be about "KV things"

I wonder if adding this info here adds too much complexity / branchiness on top of the KV snapshots stuff and is not really actionable enough 

Maybe it's enough in this section just to add the link to the 'Storage Layer > compactions' since that section also has the blurb about this env var and (for now) we can leave it out of here until we have more experience using this env var during escalations / overload situations, at which point we could document how to check "will this help anyway?"

in other words it seems reasonable to start by documenting the variable in more of a "try this out with your setup" context and remove it from this "your stuff is on fire, set this env var and see what happens" context

does that seem reasonable to you?

File: src/current/v23.2/common-issues-to-monitor.md [hunk 3]
```diff
@@ -106,6 +110,12 @@ Issues at the storage layer, including an [inverted LSM]({% link {{ page.version
 {% include {{ page.version.version }}/prod-deployment/resolution-inverted-lsm.md %}
 {{site.data.alerts.end}}
 
+{{site.data.alerts.callout_success}}
+[XXX](): <font color="red">QUESTION</font>: the instructions above say to reduce your workload concurrency, so it seems like maybe we should NOT be recommending increasing compaction concurrency here? b/c if we had IOPs and CPU headroom we would not be in this bad state, and thus increasing compaction concurrency would use *even more* IO / CPU right?  That's my sense anyway but please confirm.
```
Comment (rmloveland) [inline src/current/v23.2/common-issues-to-monitor.md:114]: @jbowens depending on your thoughts in the other convo we could remove this blurb here too for now.  Although this section does seem more "purely about storage", it seems to share the property that maybe one's first try setting this env var shouldn't be during an escalation / overload situation?

File: src/current/v23.2/cluster-setup-troubleshooting.md (hunk 2 above)
Comment (jbowens) [reply src/current/v23.2/cluster-setup-troubleshooting.md:184]: Yeah, that's reasonable. What do you think about also adding a tidbit into the inverted LSM resolution blurb saying that _if you have available CPU and disk bandwidth headroom_, you may consider increasing the compaction concurrency to reshape the LSM faster.

Comment (rmloveland) [reply src/current/v23.2/cluster-setup-troubleshooting.md]: > What do you think about also adding a tidbit into the inverted LSM resolution blurb saying that if you have available CPU and disk bandwidth headroom, you may consider increasing the compaction concurrency to reshape the LSM faster.

sounds great! I revised the inverted LSM include file to pull in the info about the new env var.  And then also rewrote the info about the new env var to mention both that it _could_ help during inverted LSMs or be used to increase overall performance (but pls test it etc).  

PTAL and let me know what you think!

**Review by jbowens — APPROVED** [review]
lgtm!

File: src/current/_includes/v23.2/storage/compaction-concurrency.md [hunk 4]
```diff
@@ -0,0 +1 @@
+By default, the [storage engine]({% link {{ page.version.version }}/architecture/storage-layer.md %}) uses a compaction concurrency of 3. If you have sufficient IOPs and CPU headroom, you can consider increasing this setting via the `COCKROACH_COMPACTION_CONCURRENCY` environment variable. This may be useful in the following scenarios: it may help to reshape the LSM faster in [inverted LSM]({% link {{ page.version.version }}/architecture/storage-layer.md %}#inverted-lsms) scenarios; and it can lead to increased overall performance for some workloads. We **strongly recommend** testing your workload against non-default values of this setting.
```
Comment (taroface) [inline src/current/_includes/v23.2/storage/compaction-concurrency.md:1]:
```suggestion
By default, the [storage engine]({% link {{ page.version.version }}/architecture/storage-layer.md %}) uses a compaction concurrency of 3. If you have sufficient IOPS and CPU headroom, you can consider increasing this setting via the `COCKROACH_COMPACTION_CONCURRENCY` environment variable. This may help to reshape the LSM more quickly in [inverted LSM]({% link {{ page.version.version }}/architecture/storage-layer.md %}#inverted-lsms) scenarios; and it can lead to increased overall performance for some workloads. Cockroach Labs **strongly recommends** testing your workload against non-default values of this setting.
```

**Review by taroface — APPROVED** [review]
LGTM with suggestions to one passage!
